crates/core/src/network/handler/lists.rs

```diff
@@ -18,26 +19,53 @@ use lib3h_protocol::{
 };
 use std::sync::Arc;
 
-#[autotrace]
-#[holochain_tracing_macros::newrelic_autotrace(HOLOCHAIN_CORE)]
-pub fn handle_get_authoring_list(get_list_data: GetListData, context: Arc<Context>) {
+//#[autotrace]
+//#[holochain_tracing_macros::newrelic_autotrace(HOLOCHAIN_CORE)]
+pub fn handle_get_authoring_list(_get_list_data: GetListData, context: Arc<Context>) {
     let c = context.clone();
     let closure = async move || {
-        let address_map = create_authoring_map(context.clone());
-
-        let action = Action::RespondAuthoringList(EntryListData {
+        // let address_map = create_authoring_map(context.clone());
+
+        // currently sim2h asks for the authoring list and treats it just the same
+        //as the gossiping list, i.e. as data that you hold.  This is a problem because
+        //it means that gossiping isn't actually working right.  So instead of fixing that in
+        //sim2h, we are doing a short term fix of just doing a fast push of re-publishing all
+        //the items we have authored instead!  This creates a one-time burst of connections to
+        // the sim2h server on join, so it's not efficient, but it works.
+        /*let action = Action::RespondAuthoringList(EntryListData {
             space_address: get_list_data.space_address,
             provider_agent_id: get_list_data.provider_agent_id,
             request_id: get_list_data.request_id,
             address_map: address_map.into(),
         });
         dispatch_action(context.action_channel(), ActionWrapper::new(action));
+        //        */
+
+        for (address, is_public) in get_all_chain_entries_with_public(context.clone()) {
+            if is_public {
+                let result = publish(address.clone(), &context).await;
+                if result.is_err() {
+                    error!(
+                        "Error publishing entry during get authoring list: {:?}",
+                        result
+                    );
+                }
+            }
+            let result = publish_header_entry(address, &context).await;
+            if result.is_err() {
+                error!(
+                    "Error publishing header during get authoring list: {:?}",
+                    result
+                );
+            }
+        }
```

Comment:
I would expect this hack to cause much more strain on the whole network since sim2h handles publishes different to how it (used to) handle(s) authoring lists. Doesn't publishes always result in store requests send to the according DHT authorities? Which would now happen every time a node comes online...

I don't think this is the right solution. If sim2h does not treat authoring lists as authoring lists anymore, that should get back in.

Reply:
We cannot have authors marked as holding the data - because they are not authoritative - the meta info will be incomplete.

Because Sim2h maintains central knowledge of who holds what, a publish should become a no-op if the item has already saturated a neighborhood.